packages/drivers/routerlicious-driver/src/documentStorageService.ts

```diff
@@ -87,13 +87,24 @@ export class DocumentStorageService implements IDocumentStorageService {
         throw new Error("NOT IMPLEMENTED!");
     }
 
-    public async createBlob(file: Uint8Array): Promise<ICreateBlobResponse> {
-        const response = this.manager.createBlob(Uint8ArrayToString(file, "base64"), "base64");
+    public async createBlob(file: ArrayBufferLike): Promise<ICreateBlobResponse> {
+        const response = this.manager.createBlob(
+            Uint8ArrayToString(
+                new Uint8Array(file), "base64"),
+            "base64");
+
         return response.then((r) => ({ id: r.sha, url: r.url }));
     }
 
-    public async readBlob(blobId: string) {
-        return IsoBuffer.from(await this.read(blobId), "base64");
+    public async readBlob(blobId: string): Promise<ArrayBufferLike> {
+        const iso = IsoBuffer.from(await this.read(blobId), "base64");
+
+        // In a Node environment, IsoBuffer may be a Node.js Buffer.  Node.js will
+        // pool multiple small Buffer instances into a single ArrayBuffer, in which
+        // case we need to slice the appropriate span of bytes.
+        return iso.byteLength === iso.buffer.byteLength
+            ? iso.buffer
+            : iso.buffer.slice(iso.byteOffset, iso.byteOffset + iso.byteLength);
```

Comment:
Last I checked, Node may bool buffers less  < 4kb into 8kb ArrayBuffers.

So if we slice, and the underlying slice copies, it'll be a small copy.  I don't feel too bad about that, given we're already allocating to decode base64, but we could potentially avoid the copy if we stopped going through IsoBuffer to decode base64.